pkg/ingester/wal.go

```diff
@@ -205,6 +205,12 @@ func (w *walWrapper) performCheckpoint() (err error) {
 		return errors.New("function to get user states not initialised")
 	}
 
+	// As the checkpoint name is taken from last WAL segment, we need to ensure
```

Comment:
Can we only cut a new segment if the current segment and the lastCheckpoint match?

Reply:
Done